crates/webidl/src/util.rs

```diff
@@ -519,31 +532,153 @@ pub enum TypePosition {
     Return,
 }
 
+trait GetArgumentPossibilities<'src> {
+    fn get_argument_possibilities(&self, record: &FirstPassRecord<'src>) -> Option<Vec<(syn::Type, String)>>;
```

Comment:
Is there a difference between a return of `Some(empty_vec)` and `None`? If so, this subtlety should be documented, if not, then we should just use a `Vec<...>` return type and canonicalize on an empty vec.

Reply:
The difference is that `None` means a conversion error (which can be filtered out using the `?` operator), while `Some(empty_vec)` is a valid result (though currently I don not think that it can be possible). This is sort of a convention that is used for all other methods that perform a conversion of types or functions.